service/tablet_allocator.cc

```diff
@@ -859,6 +1111,28 @@ class tablet_allocator_impl : public tablet_allocator::impl
         _load_balancer_stats.unregister();
     }
 
+    future<tablet_map> split_tablets(token_metadata_ptr tm, table_id table) {
+        auto& tablets = tm->tablets().get_tablet_map(table);
+
+        tablet_map new_tablets(tablets.tablet_count() * 2);
+
+        for (tablet_id tid : tablets.tablet_ids()) {
+            co_await coroutine::maybe_yield();
+
+            tablet_id new_left_tid = tablet_id(tid.value() << 1);
+            tablet_id new_right_tid = tablet_id(new_left_tid.value() + 1);
+
+            auto& tablet_info = tablets.get_tablet_info(tid);
+
+            new_tablets.set_tablet(new_left_tid, tablet_info);
+            new_tablets.set_tablet(new_right_tid, tablet_info);
```

Comment:
Let's add some comments / examples.

Before split:

tablets 0, 1, 2, 3

After split

tablets 0,1, 2,3, 4,5, 6,7

The new tablets 0,1 shares the replicas of the old tablet 0
The new tablets 2,3 shares the replicas of the old tablet 1
The new tablets 4,5 shares the replicas of the old tablet 2
The new tablets 4,5 shares the replicas of the old tablet 3

Reply:
Ok.